test/e2e/fixtures/e2e_suite.go

```diff
@@ -95,7 +95,7 @@ func (s *E2ESuite) DeleteResources() {
 
 	ctx := context.Background()
 	for _, r := range resources {
-		err := s.dynamicFor(r).DeleteCollection(ctx, metav1.DeleteOptions{PropagationPolicy: &background}, hasTestLabel)
+		err := s.dynamicFor(r).DeleteCollection(ctx, metav1.DeleteOptions{PropagationPolicy: &foreground}, hasTestLabel)
```

Comment:
I also ensured that resource deletion happens in the foreground... this should get rid of the "run e2e test two times in a row" issue where the second run would not launch pods.